sendgrid/helpers/mail/mail.py

```diff
@@ -308,14 +308,42 @@ def attachments(self):
         """
         return self._attachments
 
-    def add_attachment(self, attachment):
+    def add_attachment(self, mail_attachment):
         """Add an Attachment to this Mail.
 
         :type attachment: Attachment
         """
         if self._attachments is None:
             self._attachments = []
-        self._attachments.append(attachment)
+        if isinstance(mail_attachment, attachment.S3Attachment):
+            self.download_s3_attachment(mail_attachment)
+        self._attachments.append(mail_attachment)
+
+    def download_s3_attachment(self, s3_attachment):
+        """
+        Requires boto3, botocore to be installed
+        :param s3_attachment: S3Attachment
+        :return:
+        """
+        import boto3
+        import botocore
+        import base64
+        if s3_attachment.session is None:
+            s3 = boto3.resource('s3')
+        else:
+            s3 = s3_attachment.session.resource('s3')
+        try:
+            s3.meta.client.download_file(s3_attachment.bucket, s3_attachment.filename, s3_attachment.filename)
```

Comment:
Line length is ~111 chars (over the 79 char max for PEP8), maybe you could put the arguments on their own lines to get rid of that issue (helps with readability too)?

```python
s3.meta.client.download_file(
    s3_attachment.bucket,
    s3_attachment.filename,
    s3_attachment.filename
)

```